core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -129,22 +130,33 @@ public String toString() {
     private final AtomicBoolean closed = new AtomicBoolean(false);
     private final Map<ShardId, InternalShardLock> shardLocks = new HashMap<>();
 
-    // Setting to automatically append node id to custom data paths
-    public static final String ADD_NODE_ID_TO_CUSTOM_PATH = "node.add_id_to_custom_path";
+    /**
+     * Maximum number of data nodes that should run in an environment.
+     */
+    public static final Setting<Integer> MAX_LOCAL_STORAGE_NODES_SETTING = Setting.intSetting("node.max_local_storage_nodes", 50, 0, false,
```

Comment:
Hmmmmmmmm. I could imagine using it like `node.data:false` but since we already support that I'll just flip it to 1.